src/Workspaces/Core/Portable/Formatting/Formatter.cs

```diff
@@ -146,6 +146,14 @@ internal static async Task<Document> FormatAsync(Document document, SyntaxAnnota
             }
 
             var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            // Languages such as Razor can specify their own options instead of using the document's options.
+            var documentOptionSetProvider = document.Services.GetService<IDocumentOptionSetProvider>();
+            if (documentOptionSetProvider is not null)
```

Comment:
So why is this having to be checked _here_ rather than in the core document options logic we have, like through https://github.com/dotnet/roslyn/blob/main/src/Workspaces/Core/Portable/Options/IDocumentOptionsProvider.cs?